lib/token.js

```diff
@@ -250,7 +250,7 @@ internals.error = function (message, artifacts) {
 
 internals.b64decode = function (string) {
 
-    return Buffer.from(string, 'base64').toString('binary');
+    return Buffer.from(string, 'base64').toString();
```

Comment:
Apologies that this PR hasn't gotten the attention it deserves!  I have a question here— do you have a sense for why this latin1 encoding was used in the first place?  It stands out a little bit to me that it was going out of its way to avoid utf-8, which seems like it would have been more natural.  Mostly want to ensure there's not some security or portability issue we're creating here.

Reply:
Yeah, I looked into this and to the best guess it goes back to the `jws` module that has the lines.

```javascript
var encodedHeader = base64url(toString(header), 'binary');
var encodedPayload = base64url(toString(payload), encoding)
```

I followed the commits back they were originally both 'binary'.  When encoding was added as a feature it was only put on the `payload` and not the `header`.

My guess is that `jsonwebtoken` does not allow you to add custom fields to the header, so there was no need to use utf8, since everything standard in the header is latin1.  It was my thought that since our `jwt` allows putting custom objects in the header we should use the JavaScript standard of utf8.

If the RFC I saw this:
_Base64url encoding the bytes of the UTF-8 representation of the JWS Header yields this Encoded JWS Header value:_

Which made me believe there is nothing against the spec. 

 Either way, LMK.  I am fine with leaving it as is or removing it.  The original issue was that  we were calling `Utils.b64stringify()` with a 2nd argument, which has been fixed above.